utils/sl/check_script.py

```diff
@@ -78,51 +85,47 @@ def __init__(self, fname):
-                if re.search(r"^\"\"\"", line):
-                    scons_header = True
-                elif not re.search(r'^#', line):
+                # Insert out header after the first blank line.
+                # TODO: This should just indent at the start of the file, or the first
+                # non-docstring line.  Some files (src/bin/smd...) use a docstring on the first
+                # line and a function call on the second which breaks most simple logic here.
+                # Luckily flake8 will warn on that so maybe an acceptable cost.
+                if line.strip() == '':
+                    added = self.write_header(outfile)
+                    _remap_count()
+                    new_lineno += added
                     scons_header = True
-                if scons_header:
-                    new_lineno += self.write_header(outfile)
 
     @staticmethod
     def read_variables(outfile, prefix, variables):
         """Add code to define fake variables for pylint"""
-        newlines = 2
-        outfile.write("# pylint: disable=invalid-name\n")
+        newlines = 0
         for variable in variables:
-            outfile.write("%sprint(\"%%s\" %% str(%s))\n" % (prefix, variable))
-            newlines += 1
-        outfile.write("# pylint: enable=invalid-name\n")
+            outfile.write('# pylint: disable-next=consider-using-f-string\n')
+            outfile.write(f"{prefix}print('%s' % str({variable}))\n")
+            newlines += 2
         return newlines
 
     @staticmethod
     def write_variables(outfile, prefix, variables):
         """Add code to define fake variables for pylint"""
-        newlines = 4
-        outfile.write("# pylint: disable=invalid-name\n")
-        outfile.write("# pylint: disable=import-outside-toplevel\n")
+        newlines = 0
 
         for variable in variables:
             if variable.upper() == 'PREREQS':
-                newlines += 4
-                outfile.write("%sfrom prereq_tools import PreReqComponent\n"
-                              % prefix)
-                outfile.write("%sscons_temp_env = DefaultEnvironment()\n"
-                              % prefix)
-                outfile.write("%sscons_temp_opts = Variables()\n" % prefix)
-                outfile.write("%s%s = PreReqComponent(scons_temp_env, "
-                              "scons_temp_opts)\n" % (prefix, variable))
+                newlines += 1
+# pylint: disable-next=line-too-long
+                outfile.write(f'{prefix}{variable} = PreReqComponent(DefaultEnvironment(), Variables())\n')  # noqa: E501
                 variables.remove(variable)
         for variable in variables:
             if "ENV" in variable.upper():
                 newlines += 1
-                outfile.write("%s%s = DefaultEnvironment()\n" % (prefix,
-                                                                 variable))
+                outfile.write("%s%s = DefaultEnvironment()\n" % (prefix, variable))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string